src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
```

Comment:
Also you can create separate function with will accept person and chech is person 'man' or not
Just not to dublicate this code - p.sex === 'm' / p.sex === 'f'

const isMan = (person: YourInterface) => person.sex === 'm';

and for man use isMan(person)
for woman - !isMan(person)